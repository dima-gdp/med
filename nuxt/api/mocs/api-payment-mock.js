export const mockPaymentCreatedDataFixture = {
  id: 9,
  orderId: 1,
  userId: 32,
  invoiceId: '28ada05a-000f-5000-8000-158e10067e04',
  description: 'Оплата заказа №1 для Дмитрий Л libertyswift@yandex.ru',
  paymentMethod: 'bank_card',
  status: 'created',
  order: {
    itemTitle: 'Test',
    itemUrl: '/',
  },
  paidSum: '3000',
  paidCurrency: 'RUB',
  paidAt: null,
  confirmationUrl:
    'https://yoomoney.ru/payments/external/confirmation?orderId=28ada05a-000f-5000-8000-158e10067e04',
  createdAt: 1629195418,
  updatedAt: 1629195418,
}

export const mockCreatePaymentFixture = {
  data: {
    id: 9,
    orderId: 1,
    userId: 32,
    invoiceId: '28ada05a-000f-5000-8000-158e10067e04',
    description: 'Оплата заказа №1 для Дмитрий Л libertyswift@yandex.ru',
    paymentMethod: 'bank_card',
    status: 'created',
    order: {
      itemTitle: 'Test',
      itemUrl: '/',
    },
    paidSum: '3000',
    paidCurrency: 'RUB',
    paidAt: null,
    confirmationUrl:
      'https://yoomoney.ru/payments/external/confirmation?orderId=28ada05a-000f-5000-8000-158e10067e04',
    createdAt: 1629195418,
    updatedAt: 1629195418,
  },
}

export const mockPaymentFailedDataFixture = {
  id: 9,
  orderId: 1,
  userId: 32,
  invoiceId: '28ada05a-000f-5000-8000-158e10067e04',
  description: 'Оплата заказа №1 для Дмитрий Л libertyswift@yandex.ru',
  paymentMethod: 'bank_card',
  status: 'failed',
  order: {
    itemTitle: 'Test',
    itemUrl: '/',
  },
  paidSum: '3000',
  paidCurrency: 'RUB',
  paidAt: null,
  confirmationUrl:
    'https://yoomoney.ru/payments/external/confirmation?orderId=28ada05a-000f-5000-8000-158e10067e04',
  createdAt: 1629195418,
  updatedAt: 1629195418,
}

export const mockPaymentPaidDataFixture = {
  id: 9,
  orderId: 1,
  userId: 32,
  invoiceId: '28ada05a-000f-5000-8000-158e10067e04',
  description: 'Оплата заказа №1 для Дмитрий Л libertyswift@yandex.ru',
  paymentMethod: 'bank_card',
  status: 'paid',
  order: {
    itemTitle: 'Test',
    itemUrl: '/',
  },
  paidSum: '3000',
  paidCurrency: 'RUB',
  paidAt: null,
  confirmationUrl:
    'https://yoomoney.ru/payments/external/confirmation?orderId=28ada05a-000f-5000-8000-158e10067e04',
  createdAt: 1629195418,
  updatedAt: 1629195418,
}
