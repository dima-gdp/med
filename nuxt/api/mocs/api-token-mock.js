export const mockTokenIdSuccessFixture = {
  id: '237457',
  userId: null,
  type: 'password_reset',
  sendMethod: 'call',
  sendTo: '79115746172',
  isUsed: false,
  usedAt: null,
  expireAt: 1626345855,
  createdAt: 1626259455,
  updatedAt: 1626259455,
}

export const mockTokenIdFailFixture = {
  errors: [
    {
      source: {
        pointer: '/data/attributes/sendTo',
      },
      detail: 'Введите номер в формате 79990001111',
      status: '422',
    },
  ],
}

export const mockValidateSuccessFixture = {
  id: '237457',
  validate: true,
}

export const mockValidateFailFixture = {
  id: '237457',
  validate: false,
}

export const mockPasswordResetSuccessFixture = {
  id: '49',
  userId: 49,
}

export const mockPasswordResetFailFixture = {
  errors: [
    {
      code: '0',
      status: '500',
      title: 'Internal Server Error',
      detail: 'Возникла внутренняя ошибка сервера.',
    },
  ],
}
