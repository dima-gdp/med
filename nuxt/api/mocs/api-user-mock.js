/**
 * @module api-mocs
 */

import { Deserializer } from 'jsonapi-serializer'
const jsonApiOpts = { keyForAttribute: 'camelCase', dataLinks: {} }
const deserializer = new Deserializer(jsonApiOpts)

export const mockUserDataFixture = {
  id: 9,
  email: 'alexandr.anciferov@gmail.com',
  emailValidate: true,
  phoneValidate: true,
  external: false,
  profile: {
    userId: 9,
    positionId: 1,
    companyId: 213,
    firstName: 'Александр',
    middleName: 'Николаевич',
    lastName: 'Анциферов',
    avatar: null,
    id: '9',
  },
}

export const mockUserMenuFixture = [
  {
    type: 'panel',
    visible: true,
  },
  {
    type: 'profile',
    visible: true,
  },
  {
    type: 'subscribe',
    visible: true,
  },
  {
    type: 'certificate',
    visible: false,
  },
  {
    type: 'curator',
    visible: false,
  },
  {
    type: 'course',
    visible: false,
  },
  {
    type: 'orders',
    visible: true,
  },
  {
    type: 'logout',
    visible: true,
  },
]

export const mockPossibleMenuItemsFixture = [
  {
    type: 'panel',
    url: '/admin',
    title: 'Панель управления',
    sort: 1,
  },
  {
    type: 'profile',
    url: '/user',
    title: 'Профиль',
    sort: 2,
  },
  {
    type: 'subscribe',
    url: '/mysubscribe',
    title: 'Подписки',
    sort: 3,
  },
  {
    type: 'orders',
    url: '/orders/list',
    title: 'Мои заказы',
    sort: 7,
  },
  {
    type: 'logout',
    url: '/user/sign-in/logout',
    title: 'Выйти',
    sort: 8,
  },
]

export const mockUserAuthFixture = {
  identity: 'admin@s256.dev',
  token: 'eVLcXHtojhi3AB5tHC0wQSdAzl979qystHGTBysb',
  userId: 1,
}

export async function GET_STATS() {
  const responseData = {
    data: {
      id: 'user',
      type: 'stats',
      attributes: {
        'internal-count': 150940,
      },
    },
  }

  return {
    data: await deserializer.deserialize(responseData),
  }
}

export async function GET_USER_AND_PROFILE(userId) {
  const responseData = {
    data: {
      id: '' + userId,
      type: 'user',
      attributes: {
        id: userId,
        email: 'alexandr.anciferov@gmail.com',
        'email-validate': true,
        'phone-validate': true,
        external: false,
      },
      relationships: {
        profile: {
          data: {
            id: '' + userId,
            type: 'profile',
          },
        },
      },
    },
    included: [
      {
        id: '' + userId,
        type: 'profile',
        attributes: {
          'user-id': userId,
          'position-id': 1,
          company_id: 213,
          firstName: 'Александр',
          middleName: 'Николаевич',
          lastName: 'Анциферов',
          avatar: null,
        },
      },
    ],
  }

  return {
    data: await deserializer.deserialize(responseData),
  }
}

export async function GET_USER_MENU(userId) {
  const responseData = {
    data: {
      id: '' + userId,
      type: 'menu',
      attributes: {
        id: '' + userId,
        items: [
          {
            type: 'panel',
            visible: true,
          },
          {
            type: 'profile',
            visible: true,
          },
          {
            type: 'subscribe',
            visible: true,
          },
          {
            type: 'certificate',
            visible: true,
          },
          {
            type: 'curator',
            visible: false,
          },
          {
            type: 'course',
            visible: true,
          },
          {
            type: 'logout',
            visible: true,
          },
        ],
      },
    },
  }

  return {
    data: await deserializer.deserialize(responseData),
  }
}
