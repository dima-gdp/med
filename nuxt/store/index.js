export const state = () => ({
  currentBreakpoint: '',
})
export const mutations = {
  SET_CURRENT_BREAKPOINT(state, payload) {
    state.currentBreakpoint = payload
  },
}
