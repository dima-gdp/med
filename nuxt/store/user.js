export const state = () => ({
  isLoggedIn: false,
  userId: null,
  userToken: null,
  fromUserId: null,
  userData: {
    profile: {},
    hasPromo: false,
  },
  userDataIsLoading: false,
  menu: null,
  isUserLoaded: false,
})

export const getters = {
  hasCompany: (state) => !!state.userData?.profile?.companyId,
}

export const mutations = {
  SET_AUTH(state, userData) {
    if (userData?.userId && userData?.token) {
      const { userId, token } = userData
      state.isLoggedIn = true
      state.userId = userId
      state.userToken = token
    }
    state.fromUserId = userData?.fromUserId
  },

  RESET_AUTH(state) {
    state.isLoggedIn = false
    state.userId = null
    state.userToken = null
    state.fromUserId = null
    state.userData = { profile: {},  hasPromo: false }
  },

  SET_USER_DATA_LOADING(state, isLoading) {
    state.userDataIsLoading = isLoading
  },

  SET_USER_DATA(state, userData) {
    state.userData = userData
  },

  SET_MENU_DATA(state, userMenuData) {
    state.menu = userMenuData
  },

  SET_USER_LOADED(state) {
    state.isUserLoaded = true
  },

  SET_USER_PROMO(state, hasPromo) {
    state.userData.hasPromo = hasPromo
  },
}
