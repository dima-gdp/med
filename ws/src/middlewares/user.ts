import type {Socket as SocketIO} from 'socket.io'

import { DataSource } from '../data-source'

type Socket = SocketIO & {
  canConnect: boolean,
  user: any
}

// Этот тип на самом деле должен быть классом с геттером administrator,
// а вообще текущие роли на ws не подразумевают ситуацию, когда будет роль с частичными правами администратора трансляции (либо все либо ничего)
interface DbUser {
  user_id: number | string;
  username: string;
  role: string[];
  token: string;
  administrator: boolean
}

export default (dataSource: DataSource) => {
  return async function (socket: Socket, next: Function): Promise<any> {
    const { redis, db } = dataSource

    const queryToken = socket.handshake.query?.token
    const queryRoom = socket.handshake.query?.room

    if(typeof queryToken != 'string') throw Error('Token is not string')
    if(typeof queryRoom != 'string') throw Error('QueryRoom is not string')

    const token: string = queryToken.toString()
    const room: string = queryRoom.toString()

    let user = await redis.findUserByToken(token)

    if (user === null) {
      const [userFromDb]: DbUser[] = await db.getUserByToken(token)
        .catch((e: string) => next(new Error(e)))

      if (userFromDb) {
        userFromDb.administrator = [
          'manager',
          'editor',
          'administrator',
          'moderator',
        ].some(role => userFromDb.role.includes(role))

        redis.addUser(userFromDb, token)

        user = userFromDb
      } else {
        return new Error('Пользователь не найден')
      }
    }

    // ))
    socket.user = {
      socketId: socket.id,
      ...user,
    }

    const { user_id } = user

    const client = await redis.findUserInRoom(room, token)

    // @ts-ignore
    const [viewpoint] = client !== null && client.viewpoint ? [client.viewpoint] : await db.getViewPoint(user_id, room)

    if (viewpoint) {
      // Удалить пользователя из списка приглашённых
      if (viewpoint.parent_id && !viewpoint.invite_deleted_at) {
        await db.deleteStreamInvite(viewpoint.id)
          .catch((e: string) => {
            throw new Error(e)
          })
      }

      // Всегда обновлять данные о браузере при заходе пользователя
      await db.clearViewPointEndedField(viewpoint.id)
        .catch((e: string) => {
          throw new Error(e)
        })

      await db.clearInvitedViewPointEndedFields(viewpoint.id)
        .catch((e: string) => {
          throw new Error(e)
        })
    }
    console.log(user)
    const payload = {
      ...user,
      socketId: socket.id,
      viewpoint: viewpoint || (await db.createViewPoint(+user_id, +room))[0],
      room,
    }

    await redis.addUserToRoom(payload, token)

    return next()
  }
}
