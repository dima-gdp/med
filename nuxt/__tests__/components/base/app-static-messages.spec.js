import { shallowMount, createLocalVue } from '@vue/test-utils'
import { mockUserDataFixture } from '@/api/mocs/api-user-mock'
import Vuex from 'vuex'
import { getters as realUserModuleGetters } from '@/store/user'
import SAlert from '@/components/ui-system/s-alert'
import AppStaticMessages from '@/components/base/app-static-messages'

const localVue = createLocalVue()
localVue.use(Vuex)

describe('@/components/base/app-static-messages', () => {
  let store
  let state

  beforeEach(() => {
    state = {
      userData: {
        mockUserDataFixture,
        profile: {
          ...mockUserDataFixture.profile,
          companyId: null,
        },
      },
    }

    store = new Vuex.Store({
      modules: {
        user: {
          state,
          getters: realUserModuleGetters,
          namespaced: true,
        },
      },
    })
  })

  it('Отображается только сообщение об отсутствии организации', () => {
    const wrapper = shallowMount(AppStaticMessages, {
      store,
      localVue,
      stubs: {
        SAlert,
      },
      mocks: {
        $auth: {
          isAdminAsUser: false,
        },
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('Отображается сообщение об заходе под админом и об отсутствии организации', () => {
    const wrapper = shallowMount(AppStaticMessages, {
      store,
      localVue,
      stubs: {
        SAlert,
      },
      mocks: {
        $auth: {
          fromUserId: true,
        },
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('Закрытие сообщения при клике на кнопку его закрытия', async () => {
    const wrapper = shallowMount(AppStaticMessages, {
      store,
      localVue,
      stubs: {
        SAlert,
      },
      mocks: {
        $auth: {
          isAdminAsUser: false,
        },
      },
    })
    const alert = wrapper.findComponent(SAlert)

    expect(alert.classes()).toEqual(expect.arrayContaining(['company-alert']))
    expect(alert.exists()).toBe(true)

    await alert.vm.$emit('close')

    expect(alert.exists()).toBe(false)
  })
})

describe('@/components/base/app-static-message', () => {
  let store
  let state

  beforeEach(() => {
    state = {
      userData: {
        profile: {},
      },
    }

    store = new Vuex.Store({
      modules: {
        user: {
          state,
          getters: realUserModuleGetters,
          namespaced: true,
        },
      },
    })
  })

  it('Не отображать сообщение, если данных пользователя нету', () => {
    const wrapper = shallowMount(AppStaticMessages, {
      store,
      localVue,
      stubs: {
        SAlert,
      },
      mocks: {
        $auth: {
          isAdminAsUser: false,
        },
      },
    })

    const alert = wrapper.findComponent(SAlert)
    expect(alert.exists()).toBe(false)
  })
})
