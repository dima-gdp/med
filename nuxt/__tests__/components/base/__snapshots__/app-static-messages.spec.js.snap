// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@/components/base/app-static-messages Отображается сообщение об заходе под админом и об отсутствии организации 1`] = `
<div>
  <div>
    <div class="s-alert s-alert--info">
      <evaicon-stub name="info-outline" class="s-alert__icon"></evaicon-stub>
      <div class="s-alert__message sp-tips">
        Внимание, вы авторизованы под чужим пользователем!
      </div>
      <!---->
    </div>
  </div>
  <div class="static-alert">
    <div class="company-alert scale-up-ver-top slide-in-top">
      <div class="s-alert s-alert--warning">
        <evaicon-stub name="alert-circle-outline" class="s-alert__icon"></evaicon-stub>
        <div class="s-alert__message sp-tips"><span>
        Для того, чтобы вам стали доступны все функции сайта, заполните, пожалуйста, сведения об
        организации.
        <a href="/user" class="static-alert__link">Заполнить</a></span></div>
        <evaicon-stub name="close-outline" class="s-alert__close"></evaicon-stub>
      </div>
    </div>
  </div>
</div>
`;

exports[`@/components/base/app-static-messages Отображается только сообщение об отсутствии организации 1`] = `
<div>
  <!---->
  <div class="static-alert">
    <div class="company-alert scale-up-ver-top slide-in-top">
      <div class="s-alert s-alert--warning">
        <evaicon-stub name="alert-circle-outline" class="s-alert__icon"></evaicon-stub>
        <div class="s-alert__message sp-tips"><span>
        Для того, чтобы вам стали доступны все функции сайта, заполните, пожалуйста, сведения об
        организации.
        <a href="/user" class="static-alert__link">Заполнить</a></span></div>
        <evaicon-stub name="close-outline" class="s-alert__close"></evaicon-stub>
      </div>
    </div>
  </div>
</div>
`;
